src/components/icon-button/icon-button.component.tsx

```diff
@@ -39,6 +40,10 @@ const IconButton = React.forwardRef<HTMLButtonElement, IconButtonProps>(
     }: IconButtonProps,
     ref
   ) => {
+    const { disabled: batchSelectionDisabled } = useContext(
+      BatchSelectionContext
+    );
+    const isDisabled = disabled || batchSelectionDisabled || false;
```

Comment:
comment (non-blocking): It might be worth adopting the same approach as the other components that consume this context wrt to using default values so it's consistent etc